packages/core-utils/src/app/scheduled-task.ts

```diff
@@ -51,20 +51,16 @@ export abstract class ScheduledTask {
         throw e
       }
 
-      if (rerunImmediately) {
-        // Purposefully do not await
-        this.run()
-        return
-      }
-
-      try {
-        await sleep(this.periodMilliseconds)
-      } catch (e) {
-        logError(
-          log,
-          `Error sleeping in ScheduledTask! Continuing execution.`,
-          e
-        )
+      if (!rerunImmediately) {
+        try {
+          await sleep(this.periodMilliseconds)
+        } catch (e) {
+          logError(
+            log,
+            `Error sleeping in ScheduledTask! Continuing execution.`,
+            e
+          )
+        }
```

Comment:
Sorry have been slow to review this but I've been trying to figure out what the change actually does 😅

Where does rerun take place now? Or is it just that this didn't need to be here generally?

Reply:
The changed code is within a `while(this.running)` loop. So just jumping over the sleep makes it re-run immediately. The only material change is not churning through promises and instead continuing to loop within the same function.